src/Native/System.Security.Cryptography.Native/pal_asn1.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_asn1.h"
+
+#include <openssl/objects.h>
+
+static_assert(NID_undef == 0, "Managed code relies on NID_undef == 0");
+static_assert(NID_secp224r1 == 713, "Managed code relies on NID_secp224r1 == 713");
+static_assert(NID_secp384r1 == 715, "Managed code relies on NID_secp384r1 == 715");
+static_assert(NID_secp521r1 == 716, "Managed code relies on NID_secp521r1 == 716");
+
+extern "C" int64_t Asn1IntegerGet(const ASN1_INTEGER* a)
+{
+    return ASN1_INTEGER_get(a);
+}
+
+extern "C" const ASN1_OBJECT* ObjTxt2Obj(const char* s)
+{
+    return OBJ_txt2obj(s, true);
+}
+
+extern "C" int32_t ObjObj2Txt(char* buf, int32_t buf_len, const ASN1_OBJECT* a)
+{
+    return OBJ_obj2txt(buf, buf_len, a, true);
+}
+
+extern "C" const ASN1_OBJECT* GetFriendlyNameSharedObject(const char* friendlyName)
+{
+    int nid = OBJ_ln2nid(friendlyName);
+
+    if (nid == NID_undef)
+    {
+        nid = OBJ_sn2nid(friendlyName);
+    }
+
+    if (nid == NID_undef)
+    {
+        return nullptr;
+    }
+
+    return OBJ_nid2obj(nid);
+}
+
+extern "C" int32_t ObjSn2Nid(const char* sn)
+{
+    return OBJ_sn2nid(sn);
+}
+
+extern "C" ASN1_OBJECT* ObjNid2Obj(int32_t nid)
+{
+    return OBJ_nid2obj(nid);
+}
+
+extern "C" void Asn1ObjectFree(ASN1_OBJECT* a)
+{
+    ASN1_OBJECT_free(a);
+}
+
+extern "C" ASN1_BIT_STRING* D2IAsn1BitString(ASN1_BIT_STRING** zero, const unsigned char** ppin, int32_t len)
+{
+    return d2i_ASN1_BIT_STRING(zero, ppin, len);
+}
+
+extern "C" void Asn1BitStringFree(ASN1_STRING* a)
+{
+    ASN1_BIT_STRING_free(a);
+}
+
+extern "C" ASN1_OCTET_STRING* D2IAsn1OctetString(ASN1_OCTET_STRING** zero, const unsigned char** ppin, int32_t len)
```

Comment:
It feels like they could be done just one at a time for incremental goodness:

* d2i_ASN1_OCTET_STRING => DecodeAsn1OctetString(const unsigned char* buf, int32_t len)
* Interop.libcrypto.OpenSslD2I(Interop.libcrypto.d2i_ASN1_OCTET_STRING, data) => Interop.Crypto.DecodeAsn1OctetString(data)

But if it feels easier to tackle in phases, I'll take your word on it.  (Since it's always easier to critique than to do :)).